server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/api/DeleteCondition.java

```diff
@@ -0,0 +1,173 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.view.api;
+
+import java.util.Objects;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.james.core.MailAddress;
+import org.apache.james.queue.api.ManageableMailQueue;
+import org.apache.mailet.Mail;
+
+import com.google.common.base.Preconditions;
+
+public interface DeleteCondition {
+    boolean shouldBeDeleted(Mail mail);
+
+    class All implements DeleteCondition {
+        @Override
+        public boolean shouldBeDeleted(Mail mail) {
+            Preconditions.checkNotNull(mail);
+            return true;
+        }
+
+        @Override
+        public final boolean equals(Object obj) {
+            return obj instanceof All;
+        }
+
+        @Override
+        public final int hashCode() {
+            return Objects.hashCode(All.class);
+        }
+    }
+
+    class WithSender implements DeleteCondition {
+        private final String senderAsString;
+
+        private WithSender(String senderAsString) {
+            this.senderAsString = senderAsString;
+        }
+
+        @Override
+        public boolean shouldBeDeleted(Mail mail) {
+            Preconditions.checkNotNull(mail);
+            return mail.getSender().asString().equals(senderAsString);
```

Comment:
I'd rather promote senderAsString as a MailAddress because the other way drop business rules implemented by MailAddress